cypress/support/commands.js

```diff
@@ -0,0 +1,112 @@
+// Cypress.Commands.add('dismiss', { prevSubject: 'optional'}, (subject, options) => { ... })
+//
+//
+// -- This will overwrite an existing command --
+// Cypress.Commands.overwrite('visit', (originalFn, url, options) => { ... })
+
+import 'cypress-file-upload';
+
+Cypress.Commands.add('login',(email, password)=>{
+    cy.get('[data-cy="login-email-text-field"]').type(email);
+    cy.get('[data-cy="login-password-text-field"]').type(password);
+})
+
+Cypress.Commands.add('loginSubmit',()=>{
+    cy.get('[data-cy=login-submit-button]').click();
+})
+
+Cypress.Commands.add('logout',()=>{
+    cy.get('[data-cy=profile-settings-logout-button]').click();
+    
+})
+
+Cypress.Commands.add('msgPrompt',(msg)=>{
+    cy.get('[data-cy=toastr-message-container]').should('have.text', msg);
+})
+
+Cypress.Commands.add('uploadImg',(imgName, message)=>{
+    cy.get('[data-cy="profile-image-upload-file-field"]').attachFile(imgName);
+    cy.get('[data-cy=toastr-message-container]').should('have.text', message);
+})
+
+Cypress.Commands.add('changeImg',(imgName, message)=>{
+    cy.get('[data-cy=profile-image-upload-label]').attachFile(imgName);
+    cy.wait(2000);
+    cy.get('[data-cy=toastr-message-container]').should('have.text', message);
+})
+
+Cypress.Commands.add('pwdChange',(currentPwd, newPwd)=>{
+    cy.get('[data-cy=change-password-current-password]').type(currentPwd);
+    cy.get('[data-cy=change-password-new-password]').type(newPwd);
+    cy.get('[data-cy=change-password-confirm-new-password]').type(newPwd);
+    cy.get('[data-cy=change-password-submit-button]').click();
+    cy.get('[data-cy=toastr-message-container]').should('have.text','Password successfully changed!');
+    cy.logout();
+        
+})
+
+Cypress.Commands.add('resetPassword',(currentPwd, newPwd)=>{
+    cy.visit('https://spinkart.neetoauth.net')
+    cy.fixture("credentials").then((user) => {
+        cy.login(user.correct.email,currentPwd);
+    })
+    cy.loginSubmit();
+    cy.get('.bp3-popover-target > .relative').click();
+    cy.get('[data-cy="nav-profile-link"]').click();
+
+    cy.get('[data-cy=profile-settings-change-password-tab]').click();
+
+    cy.get('[data-cy=change-password-current-password]').type(currentPwd);
+    cy.get('[data-cy=change-password-new-password]').type(newPwd);
+    cy.get('[data-cy=change-password-confirm-new-password]').type(newPwd);
+    cy.get('[data-cy=change-password-submit-button]').click();
+    cy.get('[data-cy=toastr-message-container]').should('have.text','Password successfully changed!');
+    cy.logout();
+})
+
+Cypress.Commands.add('emailChange',(newEmail, currentPwd)=>{
+        cy.get('[data-cy=change-email-new-email-text-field]').type(newEmail);
+        cy.get('[data-cy=change-email-current-password-text-field]').type(currentPwd);
+        cy.get('[data-cy=change-email-submit-button]').click();
+        cy.get('[data-cy=toastr-message-container]').should('have.text','Email successfully changed!')
+        cy.logout();
+})
```

Comment:
We can create a custom command  if  we want to reuse that code in almost every test suite. But if the code is used in only some of the test suite, we can create a utility function and import it whenever needed.